File: app/src/main/java/com/octosync/selfimprove/HabitRepository.java
package com.octosync.selfimprove;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

public class HabitRepository {

    private HabitDao habitDao;
    private LiveData<List<Habit>> allHabits;

    public HabitRepository(Application application) {
        AppDatabase db = AppDatabase.getDatabase(application);
        habitDao = db.habitDao();
        allHabits = habitDao.getAllHabits();
    }

    public LiveData<List<Habit>> getAllHabits() {
        return allHabits;
    }

    public void insert(Habit habit) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            habitDao.insert(habit);
        });
    }

    public void update(Habit habit) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            habitDao.update(habit);
        });
    }

    public void delete(Habit habit) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            habitDao.delete(habit);
        });
    }
}

File: app/src/main/java/com/octosync/selfimprove/HabitRepository.kt
package com.octosync.selfimprove

import kotlinx.coroutines.flow.Flow

class HabitRepository(private val habitDao: HabitDao) {

    val allHabits: Flow<List<Habit>> = habitDao.getAllHabits()

    fun getHabitsForDate(date: Long): Flow<List<Habit>> {
        return habitDao.getHabitsForDate(date)
    }

    suspend fun insert(habit: Habit) {
        habitDao.insert(habit)
    }

    suspend fun update(habit: Habit) {
        habitDao.update(habit)
    }

    suspend fun delete(habit: Habit) {
        habitDao.delete(habit)
    }
}
